pkg/apis/flowcontrol/types.go

```diff
@@ -0,0 +1,282 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flowcontrol
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// These are valid flow-dingtinguisher methods.
+const (
+	// FlowDistinguisherMethodByUser works by computing flow distinguisher with the user name in the request context so
+	// that the requests from the same user will enqueued into the same queue for processing.
+	FlowDistinguisherMethodByUserType FlowDistinguisherMethodType = "ByUser"
+	// FlowDistinguisherMethodByNamespace works by computing flow distinguisher with the user requested namespace. Any
```

Comment:
Putting "user" in this comment is confusing, because the namespace in question is not an attribute of the user but rather an aspect of the object that the request accesses.  In some sense the user is making the request, but it is also true that the user is an attribute of the request, not the other way around.  Here is a suggested rewording:
```
// FlowDistinguisherMethodByNamespace specifies that the flow distinguisher is the namespace of the object
// that the request accesses.  If the object is not namespaced or the request is a non-resource request then
// the flow distinguisher is the empty string.
```

Reply:
rewording accepted